plugins/woocommerce/src/Blocks/BlockTypes/ProductButton.php

```diff
@@ -301,6 +307,50 @@ private function get_cart_item_quantities_by_product_id( $product_id ) {
 		return isset( $cart[ $product_id ] ) ? $cart[ $product_id ] : 0;
 	}
 
+	/**
+	 * Check if a product is purchasable.
+	 *
+	 * @param \WC_Product $product The product.
+	 * @return boolean The product is purchasable.
+	 */
+	private function is_product_purchasable( $product ) {
+		$is_product_purchasable = $product->is_purchasable() && $product->is_in_stock();
+
+		if ( $product->is_type( 'grouped' ) ) {
+			$grouped_product_ids = $product->get_children();
+			foreach ( $grouped_product_ids as $child ) {
+				$child_product = wc_get_product( $child );
+				if ( ! $child_product instanceof \WC_Product ) {
+					continue;
+				}
+				if ( $child_product->is_purchasable() && $child_product->is_in_stock() ) {
+					$is_product_purchasable = true;
+					break;
+				}
+			}
+		}
+
+		return $is_product_purchasable;
```

Comment:
Not a big deal, but this could be simplified with early returns. I think it makes it easier to read and it should be a bit more performant:

```suggestion
		if ( $product->is_type( 'grouped' ) ) {	
			$grouped_product_ids = $product->get_children();
			foreach ( $grouped_product_ids as $child ) {
				$child_product = wc_get_product( $child );
				if ( ! $child_product instanceof \WC_Product ) {
					continue;
				}
				if ( $child_product->is_purchasable() && $child_product->is_in_stock() ) {
					return true;
				}
			}

			return false;
		}
		
		return $product->is_purchasable() && $product->is_in_stock();
```